integration_tests/nlg_meta_eval_test.py

```diff
@@ -39,3 +40,94 @@ def test_da_cs_en(self):
+    true_data = [[1, 2, 3, 4, 5], [2, 1, 4, 5, 2], [5, 4, 3, 2, 1]]
+    pred_data = [[2, 1, 3, 4, 5], [2, 4, 5, 5, 2], [5, 3, 4, 2, 1]]
+
+    def test_sample_level_spearmanr(self):
+
+        nlg_corr_config = CorrelationNLGConfig(level="sample", func_name="spearmanr")
+        corr_metric = nlg_corr_config.to_metric()
+        stats = corr_metric.calc_stats_from_data(self.true_data, self.pred_data)
+        stats_arr = corr_metric.aggregate_stats(stats)
+        val = nlg_corr_config.to_metric().calc_metric_from_aggregate_single(stats_arr)
+        self.assertAlmostEqual(val, 0.8162952, 3)
+
+    def test_sample_level_kendalltau(self):
+
+        nlg_corr_config = CorrelationNLGConfig(level="sample", func_name="kendalltau")
+        corr_metric = nlg_corr_config.to_metric()
+        stats = corr_metric.calc_stats_from_data(self.true_data, self.pred_data)
+        stats_arr = corr_metric.aggregate_stats(stats)
+        val = nlg_corr_config.to_metric().calc_metric_from_aggregate_single(stats_arr)
+        self.assertAlmostEqual(val, 0.69046817, 3)
+
+    def test_sample_level_pearsonr(self):
+        nlg_corr_config = CorrelationNLGConfig(level="sample", func_name="pearsonr")
+        corr_metric = nlg_corr_config.to_metric()
+        stats = corr_metric.calc_stats_from_data(self.true_data, self.pred_data)
+        stats_arr = corr_metric.aggregate_stats(stats)
+        val = nlg_corr_config.to_metric().calc_metric_from_aggregate_single(stats_arr)
+        self.assertAlmostEqual(val, 0.820707397, 3)
+
+    def test_system_level_spearmanr(self):
+
+        nlg_corr_config = CorrelationNLGConfig(level="system", func_name="spearmanr")
+        corr_metric = nlg_corr_config.to_metric()
+        stats = corr_metric.calc_stats_from_data(self.true_data, self.pred_data)
+        stats_arr = corr_metric.aggregate_stats(stats)
+        val = corr_metric.calc_metric_from_aggregate(stats_arr)
+        self.assertAlmostEqual(val, 0.815789, 3)
+
+    def test_system_level_kendalltau(self):
+
+        nlg_corr_config = CorrelationNLGConfig(level="system", func_name="kendalltau")
+        corr_metric = nlg_corr_config.to_metric()
+        stats = corr_metric.calc_stats_from_data(self.true_data, self.pred_data)
+        stats_arr = corr_metric.aggregate_stats(stats)
+        val = corr_metric.calc_metric_from_aggregate(stats_arr)
+        self.assertAlmostEqual(val, 0.66666, 3)
+
+    def test_dataset_level_spearmanr(self):
+        true_data = [[1], [2], [3], [4], [5]]
+        pred_data = [[1], [2], [3], [4], [5]]
+
+        nlg_corr_config = CorrelationNLGConfig(level="dataset", func_name="spearmanr")
+        corr_metric = nlg_corr_config.to_metric()
+        stats = corr_metric.calc_stats_from_data(true_data, pred_data)
+        stats_arr = corr_metric.aggregate_stats(stats)
+        val = corr_metric.calc_metric_from_aggregate(stats_arr)
+        self.assertAlmostEqual(val, 1.0, 3)
+
+
+class MetaEvalNLGCITest(unittest.TestCase):
```

Comment:
It looks there is no tests for `level=dataset`

Reply:
there indeed is one: `test_dataset_level_spearmanr`

Reply:
It looks there are separate test classes `MetaEval{WMTDA/NLG/NLGCI}Test`. If they are testing different aspect of the functionality, you basically need to check all edge cases into every test. `test_dataset_level_spearmanr` is contained only in `MetaEvalNLGTest`

Reply:
Sounds good. I can add one in `MetaEvalNLGCITest`.